c8d: Add `--platform` flag to history, save and load (#5331)
docker/cli: docs/reference/commandline/image_history.md

f6e427c148a7: 4 weeks ago
<missing>: 4 weeks ago
```

### <a name="platform"></a> Show history for a specific platform (--platform)

The `--platform` option allows you to specify which platform-variant to show
history for if multiple platforms are present. By default, `docker history`
shows the history for the daemon's native platform or if not present, the

> first available platform.

Reviewer: the first available platform found in alphabetical order?

Reviewer: Sounds good to me if we want to define this a bit better somewhere else (or leave the road open to a better definition at a later point, without adding to user expectations)

If the local image store has multiple platform-variants of an image, the
`--platform` option selects which variant to show the history for. An error
is produced if the given platform is not present in the local image cache.

> The platform option takes the `os[/arch[/variant]]` format, for example,

Reviewer: nit
```suggestion
The platform option takes the `os[/arch[/variant]]` format, for example
```

Reviewer 2: Let me check with the grammar police (@dvdksn 🙏 ), but I think  it's recommended to always follow `for example` with a comma. Perhaps it should start with a new sentence here though (`format. For example, ...`)

Reviewer: the word man! anyway these were just my 2 cents, so feel free to merge whenever 😄

Reviewer: If the "For example" starts the sentence, I agree with having the comma after it. As is, when reading the sentence it feel strange to pause both before and after "for example", because it makes it sound like the example came before (causing some confusion, at least to myself 😄).

Definitely minutia though

Reviewer 2: Yeah; these are hard. I basically tried to avoid ending up with "3 word sentences" because that also felt a bit silly (which is why I used the comma before `example`). I updated to use a semicolon instead, but not sure if that's 100% grammatically correct either 🥹

> `linux/amd64` or `limux/arm64/v8`. Architecture and variant are optional,

Reviewer 3: ```suggestion
`linux/amd64` or `linux/arm64/v8`. Architecture and variant are optional,
```

typo! :')

Reviewer 2: DOH!!!

"off-by-one" error 😂

> and default to the daemon's native architecture if omitted.

Reviewer 3: nit:
```suggestion
and default to the daemon's native platform if omitted.
```

Or, something like `and if omitted will fallback to the daemon's defaults.`

The following examples pulls an `alpine` image for the daemon's native platform,

> and shows the history, then provides a custom platform, which fails until the

Reviewer 3: ```suggestion
shows it's history, and then provides a custom platform, which fails until the
```

> The following examples pulls an `alpine` image for the daemon's native platform,
> and shows the history, then provides a custom platform, which fails until the
> given platform-variant is pulled;

Reviewer: nit:
```suggestion
The following example: 
- pulls an `alpine` image for the daemon's native platform (because `--platform` is omitted);
- shows the history (also for the default platform, because `--platform` is omitted);
- tries to show the history for the image while providing a custom `--platform` value, which fails until the given platform-variant is pulled;
```

Reviewer 2: Oh! Forgot to reply on these; I've been going back and forth a bit if this approach would be making things better (much more in depth) or if it would be "too much".

Happy to hear from others though. Also happy to make changes along these lines in a follow up.
